text/0497-glue-l2-construct.md

```diff
@@ -7,57 +7,77 @@
 
-The glue-alpha-module already supports three of the four common types of Glue
-Jobs: Spark (ETL and Streaming), Python Shell, Ray. This RFC will add the
-more recent Flex Job. The construct also implements AWS best practice
-recommendations, such as:
-
-* use of Secrets Management for Connection JDBC strings
-* Glue Job Autoscaling
-* defaults for Glue job specification
-
-This RFC will introduce breaking changes to the existing glue-alpha-module to
-streamline the developer experience and introduce new constants and validations.
-As an opinionated construct, the Glue L2 construct will enforce
-best practices and not allow developers to create resources that use deprecated
-libraries and tool sets (e.g. deprecated versions of Python).
-
-Optional and required parameters for each job will be enforced via interface
+A Job encapsulates a script that connects to data sources, processes
+them, and then writes output to a data target. There are four types of Glue
+Jobs: Spark (ETL and Streaming), Python Shell, Ray, and Flex Jobs. Most
+of the required parameters for these jobs are common across all types,
+but there are a few differences depending on the languages supported
+and features provided by each type. For all job types, the L2 defaults
+to AWS best practice recommendations, such as:
+
+* Use of Secrets Manager for Connection JDBC strings
+* Glue job autoscaling
+* Default parameter values for Glue job creation
+
+This iteration of the L2 construct introduces breaking changes to
+the existing glue-alpha-module, but these changes streamline the developer
+experience, introduce new constants for defaults, and replacing synth-time
+validations with interface contracts for enforcement of the parameter combinations
+that Glue supports. As an opinionated construct, the Glue L2 construct does
+not allow developers to create resources that use non-current versions
+of Glue or deprecated language dependencies (e.g. deprecated versions of Python).
+As always, L1s allow you to specify a wider range of parameters if you need
+or want to use alternative configurations.
+
+Optional and required parameters for each job are enforced via interface
 rather than validation; see [Glue's public documentation](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api.html)
 for more granular details.
 
 ### Spark Jobs
 
 1. **ETL Jobs**
 
-ETL jobs supports python and Scala language. ETL job type supports G1, G2, G4
-and G8 worker type default as G2, which customer can override. It wil default to
-the best practice version of ETL 4.0, but allow developers to override to 3.0.
-We will also default to best practice enablement the following ETL features:
+ETL jobs support pySpark and Scala languages, for which there are separate but
+similar constructors. ETL jobs default to the G2 worker type, but you can
+override this default with other supported worker type values (G1, G2, G4
+and G8). ETL jobs defaults to Glue version 4.0, which you can override to 3.0.
```

Comment:
Prefer this sentence to be shown in code actually. Consider removing "which you can override to 3.0" and instead add this to your example:

```ts
glue.ScalaSparkEtlJob(this, 'SSEJ', {
  script: glue.Code.fromBucket('bucket', 'path'),
  className: 'com.example.Hi',
  role: iam.IRole,
  workerType: WorkerType.3_0 // default is WorkerType.4_0
```

Rationale: our examples travel farther than our words. As part of our release process, we search for all constructs to see if they have an inline code example, and if not, we try to source them from the readme. So this example would be a valid option to be the code example for `glue.ScalaSparkEtlJob` and could show up in the cdk docs for the construct. We don't do any magic for our prose, so that will definitely only be in the readme.

This is just one example, but if you see opportunity to improve examples elsewhere, I encourage you to take it with the knowledge that examples end up having more influence.